pkg/operator/ceph/file/subvolumegroup/controller.go

```diff
@@ -187,6 +187,25 @@ func (r *ReconcileCephFilesystemSubVolumeGroup) reconcile(request reconcile.Requ
 		if cephCluster.Spec.External.Enable {
 			logger.Warningf("external subvolume group %q deletion is not supported, delete it manually", namespacedName)
 		} else {
+			if opcontroller.IsCleanupRequired(cephFilesystemSubVolumeGroup.GetAnnotations()) {
```

Comment:
How about moving this whole block to a helper method? This method is so long already, it helps for code readability.